### DISCUSSION TITLE
over-parallelism behavior?

### DISCUSSION THREAD

**Discussion by tooptoop4** [argoproj/argo-workflows#7482] [Q&A] [answered]
https://github.com/argoproj/argo-workflows/blob/v3.2.6/docs/workflow-controller-configmap.yaml#L18 shows parallelism setting of 10. So I understand 10 workflows can run at same time. What happens if I submit 200 workflows? do the other 190 get discarded/errored or do they queue up forever until some of the first 10 workflows finish then the 190 start running?

**Comment by sarabala1979** [comment, marked as answer]
190 will be queued and executed based creation time and priority

**Reply by tooptoop4** [reply]
@sarabala1979 is there a limit to the queue? can 1000000 workflows be queued? just trying to work out the failure points/limits

**Comment by sarabala1979** [comment]
There is no limit. As long as,  your k8s master and Etcd is able to handle the N of objects and you need to increase the memory size for workflow-controller pod.

**Reply by domderen** [reply]
@sarabala1979 I'm wondering if there were any discussions about offloading those queued workflows from workflow-controller's memory into some other mechanism. I'm in a situation similar to the one described in this thread, where I'm queuing let's say 100k workflows, but I only allow up to 1k to run in parallel. That means that 99k are using memory inside workflow-controller, but are not actually running, and could be offloaded to some disk queue to decrease memory requirements for workflow-controller. 

I'm wondering if you heard about anyone running into similar problem, or if there were any discussions about potentially offloading those queued workflows out of controller's memory?

Thanks in advance for your help!

**Reply by tooptoop4** [reply]
I'm getting https://github.com/argoproj/argo-workflows/issues/10441 issue because etcd is being overwhelmed :(
